sstables/sstables_manager.cc

```diff
@@ -330,6 +330,9 @@ std::vector<std::filesystem::path> sstables_manager::get_local_directories(const
 }
 
 void sstables_manager::on_unlink(sstable* sst) {
+    // Reclaim memory used by the sstable components and update the tracker
+    auto memory_reclaimed = sst->reclaim_memory_from_components();
+    _total_reclaimable_memory -= memory_reclaimed;
```

Comment:
I think we should just 0 both `_total_reclaimable_memory` and `_total_memory_reclaimed` here, just to be on the safe side.

Reply:
this is in sstables_manager, so global stats. @lkshminarayanan don't we have to adjust _total_memory_reclaimed here? since it's decreased in deactive().


I think we can extract
```
    auto memory_reclaimed = sst_with_max_memory->reclaim_memory_from_components();
    _total_memory_reclaimed += memory_reclaimed;
    _total_reclaimable_memory -= memory_reclaimed;
```
into a function and reuse it here.

Reply:
you're right. I'll update the patch.